webroot/js/websocket.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Websocket.
+ * This is a standalone websocket interface that passes messages to `listeners`, and accepts
+ * messages to be sent via `send`.
+ * If you have a component that would like to be notified about websocket activity you can
+ * pass it to `addListener(object)` and implement any of the following methods:
+ * 
+ * rawWebsocketMessageReceived(msg): Will send all raw JSON objects that come across the socket.
+ * websocketConnected/websocketDisconnected: Be notified on the connectivity state of the socket.
+ * 
+ * This class should not know anything about views.  Note that error states are not being passed to
+ * listeners because listeners shouldn't know or care about specific socket-level errors.
+ */
+
+import SOCKET_MESSAGE_TYPES from './chat/socketMessageTypes.js';
+
+const LOCAL_TEST = window.location.href.indexOf('localhost:') >= 0;
```

Comment:
Why not keep this one in utils.js?  Look repeated across files.

Reply:
That's the thing Safari had issues with.  Once I moved to a module then Safari didn't have those values available globally anymore.  But it was ok in Chrome.  If we do a refactor with modules then we can move them to a constants module or whatever you prefer so they're not duplicated.

Also we should get rid of local test.